src/domain/registrations/actionsDropdown/ActionsDropdown.tsx

```diff
@@ -0,0 +1,112 @@
+import { IconMenuDots } from 'hds-react';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useLocation } from 'react-router';
+
+import MenuDropdown from '../../../common/components/menuDropdown/MenuDropdown';
+import { MenuItemOptionProps } from '../../../common/components/menuDropdown/MenuItem';
+import { ROUTES } from '../../../constants';
+import { Registration } from '../../../generated/graphql';
+import useLocale from '../../../hooks/useLocale';
+import { copyRegistrationToSessionStorage } from '../../registration/utils';
+import { REGISTRATION_EDIT_ACTIONS } from '../constants';
+import {
+  addParamsToRegistrationQueryString,
+  getEditButtonProps,
+  getRegistrationFields,
+} from '../utils';
+import styles from './actionsDropdown.module.scss';
+
+export interface ActionsDropdownProps {
+  className?: string;
+  registration: Registration;
+}
+
+const ActionsDropdown = React.forwardRef<HTMLDivElement, ActionsDropdownProps>(
+  ({ className, registration }, ref) => {
+    const { t } = useTranslation();
+    const locale = useLocale();
+    const history = useHistory();
+    const { pathname, search } = useLocation();
+    const { id, registrationUrl } = getRegistrationFields(registration, locale);
+
+    const goToEditRegistrationPage = () => {
+      const queryString = addParamsToRegistrationQueryString(search, {
+        returnPath: pathname,
+      });
```

Comment:
This exact function call seems to be in many places, maybe there helper hook that return query string with return path:

`const queryStringWithReturnPath = useQueryStringWithReturnPath()`

```
const useQueryStringWithReturnPath = () => {
   const {pathname, search} = useLocation();
   return React.useMemo(() => addParamsToRegistrationQueryString(search, {
        returnPath: pathname,
    }), [pathname, search]);
}
```

Or maybe some other reusable solution :)